### PR TITLE
🍄 added argocd jenkins-slave 🍄

### PR DESCRIPTION
**Pull request by eformat**
#### What is this PR About?
add argocd slave for scaffolding argocd projects from jenkins pipelines

#### How do we test this?
see jenkins-slave-argocd/Jenkinsfile.test

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

**Comment by eformat** [comment]
> Thank you @eformat
> It seems like you've somehow added your files into a new directory `a`, can you please move them to the correct location?
> Also could you update our CI tests with this new addition? You'd need to update
> 
> * `jenkins_slaves`
> * `test_pipelines`
> * `object: test-pipelines`
>   in [.applier/group_vars/seed-hosts.yml](/redhat-cop/containers-quickstarts/blob/master/.applier/group_vars/seed-hosts.yml)

Doh! thanks @pabrahamsson , patch -p0 mess up .. fixed it up now and added test pipeline. cheers

File: jenkins-slaves/jenkins-slave-argocd/Jenkinsfile.test [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                name 'jnlp'
+                image "image-registry.openshift-image-registry.svc:5000/openshift/jenkins-slave-argocd"
+                alwaysPullImage true
+                workingDir '/tmp'
+                args '${computer.jnlpmac} ${computer.name}'
+                command ''
+                ttyEnabled false
+            }
+        }
+    }
```
Comment (pabrahamsson) [inline jenkins-slaves/jenkins-slave-argocd/Jenkinsfile.test]: I believe you can shorten this to
```
pipeline {
    agent {
        label 'jenkins-slave-argocd'
    }
}
```
What you have now works, just curious if there's a particular reason for it?

File: jenkins-slaves/jenkins-slave-argocd/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM quay.io/openshift/origin-jenkins-agent-base:4.2
+
+ARG VERSION=1.3.6
```
Comment (pabrahamsson) [inline jenkins-slaves/jenkins-slave-argocd/Dockerfile]: Can we bump this to `1.4.2`?

**Comment by eformat** [comment]
no probs, made changes now. yeah, the long version of test agent is not required. have bumped version and added {}. cheers

/retest
